synapse/handlers/e2e_keys.py

```diff
@@ -168,32 +168,36 @@ async def query_devices(
                 ) = await self.store.get_user_devices_from_cache(query_list)
                 for user_id, devices in remote_results.items():
                     user_devices = results.setdefault(user_id, {})
-                    for device_id, device in devices.items():
-                        keys = device.get("keys", None)
-                        device_display_name = device.get("device_display_name", None)
-                        if keys:
-                            result = dict(keys)
-                            unsigned = result.setdefault("unsigned", {})
-                            if device_display_name:
-                                unsigned["device_display_name"] = device_display_name
-                            user_devices[device_id] = result
 
-                # check for missing cross-signing keys.
-                for user_id in remote_queries.keys():
                     cached_cross_master = user_id in cross_signing_keys["master_keys"]
-                    cached_cross_selfsigning = (
-                        user_id in cross_signing_keys["self_signing_keys"]
+                    self_sig_key = cross_signing_keys["self_signing_keys"].get(
+                        user_id, {}
                     )
+                    cached_cross_selfsigning = bool(self_sig_key)
```

Comment:
I'm not sure if I'm missing something, but it looks like this logic is identical to the previous version?

Reply:
+1. I found the original easier to understand (and has the advantage of not introducing yet another local variable to this massive function).

Reply:
The logic is not quite identical: the proposed code would behave differently if `user_id` was present in the self-signing keys dict but set to a falsey value, like an empty dict.

I wonder if this might more clearly express intent:

```suggestion
                    cached_cross_selfsigning = bool(cross_signing_keys["self_signing_keys"].get(user_id)
```

The question then becomes: are there scenarios where this could happen, and would we be better off preventing _those_ instead? (E.g., stripping falsey values from the dicts before testing for membership)

Reply:
From looking at `get_cross_signing_keys_from_cache()`, it may be possible for a falsey value to end up in `cross_signing_keys["self_signing_keys"]`.

There's a call tree that's basically:

- `get_cross_signing_keys_from_cache()`
    - `store.get_e2e_cross_signing_keys_bulk()` (Docstring: "If a user's cross-signing keys were not found, either their user ID will not be in the dict, __or their user ID will map to None.__")
        - `store._get_bare_e2e_cross_signing_keys_bulk_txn()` (Docstring: "If a user's cross-signing keys were not found, __their user ID will not be in the dict.__")

So worth figuring out what's actually true. Or being defensive here.

(Also worth seeing if this code survived Erik's refactor in the related PR)

Reply:
The same code is present in the current version of `e2e_keys.py`, so still worth digging into.